### PR TITLE
Code update

### PR DESCRIPTION
**Pull request by Gjacquenot**
This pull request includes:

- double algorithm with correct use of math functions
- add `const` whenever possible
- use of `for (x : vector)` syntax
- add tests with google-test

### CODE REVIEW

File: numeric.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+{
+    // the machine epsilon has to be scaled to the magnitude of the values used
+    // and multiplied by the desired precision in ULPs (units in the last place)
+    return std::abs(x-y) <= std::numeric_limits<T>::epsilon() * std::abs(x+y) * ulp
+    // unless the result is subnormal
+           || std::abs(x-y) < std::numeric_limits<T>::min();
+}
+
+template<class T>
+T half(T x){}
```
Comment (Bl4ckb0ne) [inline numeric.h]: A call to this function should be stopped by a [static_assert](http://en.cppreference.com/w/cpp/language/static_assert)

File: .gitmodules [hunk 2]
```diff
@@ -0,0 +1,3 @@
+[submodule "googletest"]
```
Comment (Bl4ckb0ne) [inline .gitmodules]: Why use a whole submodule for two tests? [Catch2](https://github.com/catchorg/Catch2) would be a better fit to stay small.

File: numeric.h [hunk 3]
```diff
@@ -0,0 +1,30 @@
+#ifndef H_NUMERIC
+#define H_NUMERIC
+
+#include <cmath>
+#include <limits>
+// #include <iomanip>
```
Comment (Bl4ckb0ne) [inline numeric.h]: Clean the unused includes

File: numeric.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// #include <type_traits>
+// #include <algorithm>
+
+template<class T>
+typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type
+    almost_equal(T x, T y, int ulp=2)
```
Comment (Bl4ckb0ne) [inline numeric.h]: Cite your [sources](http://en.cppreference.com/w/cpp/types/numeric_limits/epsilon)

File: README.md [hunk 5]
```diff
@@ -1,7 +1,11 @@
 # delaunay-triangulation
 
+[![Build Status](https://travis-ci.org/Gjacquenot/delaunay-triangulation.svg?branch=master)](https://travis-ci.org/Gjacquenot/delaunay-triangulation)
```
Comment (Bl4ckb0ne) [inline README.md]: Does this needs to be changed if merged into master ?

File: numeric.h (hunk 3 above)
Comment (Gjacquenot) [reply numeric.h]: Done!

File: README.md (hunk 5 above)
Comment (Gjacquenot) [reply README.md]: Yes, once merged you need to create a travis-ci account and update the address to match your account.

**Comment by Gjacquenot** [comment]
@Bl4ckb0ne  : I have added two performance tests: with google test, I monitor computation times. If you want, I can switch to catch2 even if you don't know it.

**Comment by Bl4ckb0ne** [comment]
@Gjacquenot what's troubling me the most is the submodule dependency inside of the repo. Does it needs a specific version of google-test or can it rely on the user's setup?

If not, catch2 is lighter than google-test. I can write the test if you don't want to.

**Comment by Gjacquenot** [comment]
@Bl4ckb0ne : It does not need a specific version of google-test. I have chosen the latest release. But any would work, and a user one can also be a solution. I look at catch2

**Comment by Gjacquenot** [comment]
@Bl4ckb0ne : I have updated code to use catch.hpp instead of google test.

**Comment by Bl4ckb0ne** [comment]
Looks good to me. Anything you want to add to this PR @Gjacquenot ?

Out of curiosity, how did you find the project and why the enthusiasm to improve it? Are you using it in a personal/professional project?

Thanks a lot for your work.

**Comment by Gjacquenot** [comment]
I am looking at your project for personal interest, with a possible professional application. It is easy to look at the code and easy to use. I have also tested `qhull`, which seems much quicker but much more complicated to get started with...

**Comment by Gjacquenot** [comment]
Thanks for merging. You can now update the build badge:

- associate your GitHub account with travis to access CI service
- update the URL address in the markdown to point to the correct build

**Comment by Bl4ckb0ne** [comment]
I took care of that when I merged the PR, thanks.

**Pull request merged**